### PR TITLE
build: Updated Koin to 3.5.0 and related dependencies

### PR DESCRIPTION
**Pull request by Slava-Miligoulo-Target**
Addresses [issue](https://github.com/kotest/kotest/issues/3695)

Removed non-supported targets and updated koin to latest version. 

*Note - watchosArm32() was removed from koin.  
Koin 3.4.3...3.5.0 - [diff](https://github.com/InsertKoinIO/koin/compare/ktor-3.4.3...3.5.0)
example file - [core/koin-core-coroutines/build.gradle](https://github.com/InsertKoinIO/koin/blob/6d8b6432a1c413dfe4975f097094f42ca2efcfcb/core/koin-core-coroutines/build.gradle)

Updates:
koin - 3.5.0
kotlin - 1.9.10
gradle - 8.3
mockk - 1.13.7
Java - 11

### CODE REVIEW

**Comment by sksamuel** [comment]
Build is failing. I think latest gradle it doesn't like. I would stick with gradle 7.6

File: build.gradle.kts [hunk 1]
```diff
@@ -35,9 +35,10 @@ kotlin {
       }
 
       linuxX64()
+//      linuxArm64()  //
```
Comment (Slava-Miligoulo-Target) [inline build.gradle.kts]: @sksamuel I think is required for linuxArm64, but we'd need Koin project to add this into their build targets as well.

**Comment by Slava-Miligoulo-Target** [comment]
It may be related to a missing build target for linuxArm64. I've left a comment at the section.
But to be certain also pushed out a 7.6.2 gradle commit.  Could we run the workflow and to see if it's related to gradle version?

**Comment by Slava-Miligoulo-Target** [comment]
After reviewing the workflow errors, it does look to be related to missing build target.  Koin doesn't have it included on their end yet.

**Comment by sksamuel** [comment]
I'll release without linuxArm64 for now

**Comment by sksamuel** [comment]
I guess iosArm32TestKlibrary is missing too.

**Pull request merged**